### PR TITLE
(fix & imp) [SD-2362] Sign off shows first three letters first time then full afterwards

### PR DESCRIPTION
**Pull request by sivakuna-aap**

### CODE REVIEW

**Comment by akintolga** [comment]
:+1:

File: client/app/scripts/superdesk-users/users.js [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
         usersService.usernamePattern = /^[A-Za-z0-9_.'-]+$/;
         usersService.phonePattern = /^(?:(?:0?[1-9][0-9]{8})|(?:(?:\+|00)[1-9][0-9]{9,11}))$/;
+        usersService.signOffPattern = /^[a-zA-Z0-9]+$/;
```
Comment (plamut) [inline client/app/scripts/superdesk-users/users.js]: A shortcut would be `/^[^\W_]+$/`.

File: server/apps/users/services.py [hunk 2]
```diff
@@ -102,6 +103,29 @@ def compare_preferences(original, updates):
...
+def set_sign_off(user):
+    """
+    Set sign_off property on user if it's not set already.
+    """
+
+    if SIGN_OFF not in user:
+        if 'first_name' not in user or 'last_name' not in user:
+            user[SIGN_OFF] = user['username'][:3].upper()
+        else:
+            user[SIGN_OFF] = '%s%s' % (user['first_name'][0], user['last_name'][0])
```
Comment (plamut) [inline server/apps/users/services.py]: I find `{first_name[0]}{last_name[0]}'.format(**user)` slightly more readable, although it's just my personal opinion, thus a change is completely at your discretion.

File: server/apps/users/services.py [hunk 3]
```diff
@@ -102,6 +103,29 @@ def compare_preferences(original, updates):
...
+
+def get_sign_off(user):
+    """
+    Gets sign_off property on user if it's not set already.
+    """
+
+    if SIGN_OFF not in user:
+        set_sign_off(user)
+
+    return user[SIGN_OFF]
```
Comment (plamut) [inline server/apps/users/services.py]: Does `set_sign_off()` set the `SIGN_OFF` key?
(if not, this might throw a `KeyError`)

**Comment by plamut** [comment]
I think it's good. :+1:

File: server/apps/users/services.py (hunk 3 above)
Comment (sivakuna-aap) [inline server/apps/users/services.py]: Yes, `set_sign_off()` will set `sign_off` property

Comment (plamut) [inline server/apps/users/services.py]: OK, good then! :+1:

**Comment by mdhaman** [comment]
:+1:

**Comment by marwoodandrew** [comment]
:+1:

**Pull request merged**